aspnetcore/host-and-deploy/health-checks/samples/8.x/HealthChecksSample/Snippets/Program.cs

```diff
@@ -26,6 +26,31 @@ public static void MapHealthChecksComplete(string[] args)
         // </snippet_MapHealthChecksComplete>
     }
 
+    public static void MapHealthChecksComplete2(string[] args)
+    {
+        // <snippet_MapHealthChecksComplete2>
+        var builder = WebApplication.CreateBuilder(args);
+
+        builder.Services.AddHealthChecks()
+           .Add(new HealthCheckRegistration(
+               name: "SampleHealthCheck1",
+               instance: new DelegateHealthCheck(_ => Task.FromResult(HealthCheckResult.Healthy(HealthyMessage))),
```

Comment:
Error	CS0122	'DelegateHealthCheck' is inaccessible due to its protection level
cc @francedot @JuergenGutsch

Reply:
Thanks for the catch. Instead of DelegateHealthCheck, you should be able to pass on a custom implementation of IHealthCheck as instance.

Reply:
The best way to add a delegate as a health check is to just use AddCheck or AddAsyncCheck defined in the HealthChecksBuilderDelegateExtensions:

```
    public static void MapHealthChecksComplete2(string[] args)
    {
        // <snippet_MapHealthChecksComplete2>
        var builder = WebApplication.CreateBuilder(args);

        builder.Services.AddHealthChecks()
           .AddAsyncCheck(name: "SampleHealthCheck1",
               check: _ => Task.FromResult(HealthCheckResult.Healthy("HealthyMessage")),
               tags: null,
               timeout: default);

        var app = builder.Build();

        app.MapHealthChecks("/healthz");

        app.Run();
        // </snippet_MapHealthChecksComplete2>
    }
```

Reply:
Can I push directly into the hc2 branch? If not the method shown above should work

Reply:
What @francedot proposed:

```
 public static void MapHealthChecksComplete2(string[] args)
 {
     // <snippet_MapHealthChecksComplete2>
     var builder = WebApplication.CreateBuilder(args);

     builder.Services.AddHealthChecks()
        .Add(new HealthCheckRegistration(
            name: "SampleHealthCheck1",
            instance: new SampleHealthCheck(),
            failureStatus: null,
            tags: null,
            timeout: default)
{
    Delay = TimeSpan.FromSeconds(40),
    Period = TimeSpan.FromSeconds(30)
});

     var app = builder.Build();

     app.MapHealthChecks("/healthz");

     app.Run();
     // </snippet_MapHealthChecksComplete2>
 }
```